src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -513,7 +513,8 @@ static bool simplify_gather_shapeof(shared_ptr<Node> node)
 
     auto zero_axis = op::Constant::create<int64_t>(element::i64, Shape{}, {0});
     shared_ptr<Node> replace_node;
-    auto new_shapeof = make_shared<opset3::ShapeOf>(gather->input_value(0).get_node_shared_ptr());
+    auto new_shapeof =
+        make_shared<opset3::ShapeOf>(gather->input_value(0).get_node_shared_ptr()->output(0));
```

Comment:
This should be `gather->input_value(0)`. The input to `gather` isn't necessarily coming from output 0.

Reply:
I am pretty sure the case in lines 520-524 is wrong. The axis isn't removed from the shape, it is a 0 in the shape, and the shape can be a constant, no need for gather,

Reply:
this turns gather->shapeof into shapeof->gather (as shapes can be dynamic, we use shapeof , and gather helps with removing axis dimension ) .